flo_ai/common/flo_logger.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+from logging.handlers import RotatingFileHandler
+from typing import Optional
+
+
+class FloLogger:
+    _instances = {}
+
+    def __new__(cls, name: str, level: str = "INFO", use_file: bool = True, file_path: str = "flo.log", max_bytes: int = 1048576):
+        """
+        Custom new method to check for existing instance.
+        """
+        if name not in cls._instances:
+            instance = super().__new__(cls)  
+            cls._instances[name] = instance
+        return cls._instances[name]
+
+    def __init__(self, name: str, level: str = "INFO", use_file: bool = True, file_path: str = "flo.log", max_bytes: int = 1048576):
+
+        self.logger = logging.getLogger(name)
+        if not self.logger.handlers:
+            self.set_level(level)
+            self._setup_handler(use_file, file_path,max_bytes)
+
+    def _setup_handler(self, use_file: bool, file_path: str, max_bytes: int):
+        handler = logging.StreamHandler()
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        handler.setFormatter(formatter)
+        self.logger.addHandler(handler)
+
+        if use_file:
+            log_file_path = os.path.join(os.path.dirname(__file__), file_path)
+            file_handler = RotatingFileHandler(
+                log_file_path, 
+                maxBytes=max_bytes, 
+            )
+            file_handler.setFormatter(formatter)
+            self.logger.addHandler(file_handler)
+
+    def set_level(self, level):
+        if isinstance(level, str):
+            level = getattr(logging, level.upper(), logging.INFO)
+        self.logger.setLevel(level)
+
+    def debug(self, message: str):
+        self.logger.debug(message)
+
+    def info(self, message: str):
+        self.logger.info(message)
+
+    def warning(self, message: str):
+        self.logger.warning(message)
+
+    def error(self, message: str):
+        self.logger.error(message)
+
+    def critical(self, message: str):
+        self.logger.critical(message)
+
+
+def get_logger(name: str, level: Optional[str] = 'INFO', use_file: bool = True, file_path: str = "flo.log") -> FloLogger:
```

Comment:
Create a common set of logger, and re-use them, lets not create new logger for everything. Maybe on for CommonLogs another for Builder logs etc